index.tt

```diff
@@ -6,152 +6,349 @@
+16.7.9
+(nix-shell)<span class="nix-command">yarn --version</span>
+1.22.4</pre>
+        <p>
+          Commit the above <code>shell.nix</code> file and let you coworkers have
+          easier time setting their development environment.
+        </p>
+
       </div>
     </div>
-  </div></div>
-  
-  <hr />
-  
-  <h2>More about NixOS</h2>
-  
-  <div class="row-fluid more-about">
-    <div class="span4">
-      <h3>Nixpkgs</h3>
-      <p>
-        The Nix Packages collection (Nixpkgs) is the ever growing
-        packages collection for the Nix package manager, released
-        under a permissive MIT/X11 license.
-      </p>
-      <a class="learn-more" href="[%root%]nixpkgs">
-        More about Nixpkgs
-        <i class="fa fa-angle-right"></i>
-      </a>
-    </div>
-    <div class="span4">
-      <h3>NixOps</h3>
-      <p>
-        NixOps is a tool for deploying sets of NixOS Linux machines,
-        bare-metal or to the cloud. It extends NixOS’s declarative
-        approach to system configuration management to networks and
-        provisioning.
-      </p>
-      <a class="learn-more" href="https://github.com/NixOS/nixops">
-        More about NixOps
-        <i class="fa fa-angle-right"></i>
-      </a>
+  </section>
+  <section class="tab-pane" id="build">
+    <div class="row-fluid">
+      <div class="span6">
+        <h3>Minimal docker image</h3>
+        <p>
+          Declarative way to build minimal docker images. No build tools inside
+          docker image, no complex multi stage build process, only what your
+          application needs.
+        </p>
+        <p>
+          The following Nix expression (<code>default.nix</code>) defines a
+          docker image with <strong>only</strong> Python 3 installed in it.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.dockerTools.buildLayeredImage {
+  name = "only-python";
+  contents = [ pkgs.python3 ];
+}</pre>
+        <p>To build and run the image you need to:</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-build</strong></span>
+...
+/nix/store/hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3-docker-image-only-python.tar.gz
+<span class="nix-command">docker load -u ./result</span>
+...
+Loaded image: hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3
+<span class="nix-command">docker run hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3 python -c "print('hello world')"</span>
+hello world</pre>
+        <p>
+          Learn more <a href="[%root%]nixpkgs/manual/#sec-pkgs-dockerTools">how
+            to build docker images</a>.
+        </p>
+
+      </div>
+      <div class="span6">
+
+        <h3>Declarative cloud images</h3>
+        <p>How hard would it be to build and configure <strong>Amazon AMI?</strong></p>
+        <p>
+          With the following <code>amazon.nix</code> we defined nginx which is
+          serving example <code>/var/www</code> folder, having a valid ssl
```

Comment:
to make this more like a real world example: how would `/var/www` get content?

upload via ssh when it runs?

can we improve the experience with this example, so when someone actually does it, they should be rewarded with some website loading over https. maybe NixOS manual? or there is that nginx default page, use that as it's the same on ubuntu server after installing nginx with apt

Reply:
:+1:  i will fix this.